dnsapi/dns_octodns.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env sh
+
+# This API is a wrapper for other API so that you
+# are able to put your TXT record to multiple providers.
+# The functionality is in the responsibility of the respective API
+# therefore please check if your used APIs work.
+# e.g. for using NS1 and AWS Route53, use:
+# OCTODNS_PROVIDERS=nsone_aws
+# --dns octodns
+#
+# Author: Josef Vogt
+#
+########  Public functions #####################
+dns_octodns_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _info "Creating DNS entries via octoDNS"
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+  if [ -z "$OCTODNS_PROVIDERS" ]; then
+    OCTODNS_PROVIDERS=""
+    _err "You didn't specify OCTODNS_PROVIDERS yet."
+    _err "Please specifiy your providers and try again."
+    return 1
+  fi
+
+  #save the providers list to the account conf file.
+  _saveaccountconf OCTODNS_PROVIDERS "$OCTODNS_PROVIDERS"
+
+  for element in $(echo "$OCTODNS_PROVIDERS" | tr "_" ' '); do
+    _debug element "$element"
+    sourcecommand="$_SUB_FOLDER_DNSAPI/dns_${element}.sh"
```

Comment:
don't need to load the source file.
When you come here,  you are already loaded.

Reply:
We call the other providers from dns_octodns.sh directly and not from acme.sh. And in dns_octodns.sh, the other provider scripts are not sourced (I remember having errors when not sourcing them here). 
So the idea is to source only the the API files for the providers that are specified. 
See also my next comment.

Reply:
why not use the function `_findhook()` to search for the dns api:
https://github.com/acmesh-official/acme.sh/blob/2a2d556551e9266a3924da205c1ede55d89a689d/acme.sh#L4578